### PR TITLE
Enable weightless caching with compile_model() API without providing weights_path property

### PR DESCRIPTION
**Pull request by tkrupa-intel**
Requested in https://jira.devtools.intel.com/browse/CVS-126606?focusedId=25564110&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-25564110

### CODE REVIEW

File: src/inference/src/dev/core_impl.cpp [hunk 1]
```diff
@@ -1447,6 +1447,19 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
 
                 ov::AnyMap update_config = config;
                 update_config[ov::loaded_from_cache.name()] = true;
+
+                if (update_config[ov::cache_mode.name()] == ov::CacheMode::OPTIMIZE_SIZE) {
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.cpp:1451]: cache mode can be globally set before using `core.set_property(GPU, cache_mode(OPTIMIZE_SIZE))` and not present in current `config` / `update_config`

File: src/inference/src/dev/core_impl.cpp [hunk 2]
```diff
@@ -1447,6 +1447,19 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
...
+                    std::string weights_path;
+                    auto pos = cacheContent.modelPath.rfind('.');
+                    if (pos != cacheContent.modelPath.npos) {
+                        weights_path = cacheContent.modelPath.substr(0, pos);
+                    }
+                    weights_path += ".bin";
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.cpp:1457]: what if it's `onnx` model ?

File: src/inference/src/dev/core_impl.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/inference/src/dev/core_impl.cpp:1451]: probably, we can unconditionally set weight_path to all plugins and just add checks in each plugin to ignore this property if it's not handled by plugin

Comment (tkrupa-intel) [reply src/inference/src/dev/core_impl.cpp]: Done. Makes sense, if OPTIMIZE_SIZE is not set, weights_path will be ignored even if it's set here.

File: src/inference/src/dev/core_impl.cpp (hunk 2 above)
Comment (tkrupa-intel) [reply src/inference/src/dev/core_impl.cpp]: Changed.

File: src/plugins/intel_gpu/src/plugin/plugin.cpp [hunk 3]
```diff
@@ -596,7 +596,8 @@ std::vector<ov::PropertyName> Plugin::get_supported_properties() const {
         ov::PropertyName{ov::hint::enable_cpu_pinning.name(), PropertyMutability::RW},
         ov::PropertyName{ov::device::id.name(), PropertyMutability::RW},
         ov::PropertyName{ov::hint::dynamic_quantization_group_size.name(), PropertyMutability::RW},
-        ov::PropertyName{ov::hint::activations_scale_factor.name(), PropertyMutability::RW}
+        ov::PropertyName{ov::hint::activations_scale_factor.name(), PropertyMutability::RW},
+        ov::PropertyName{ov::weights_path.name(), PropertyMutability::RW},
```
Comment (ilya-lavrenov) [inline src/plugins/intel_gpu/src/plugin/plugin.cpp:600]: BTW, should not it be WO ? because we don't expect to query this value from plugin

Comment (tkrupa-intel) [reply src/plugins/intel_gpu/src/plugin/plugin.cpp]: Changed.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by p-durandin** [comment]
build_jenkins

**Comment by p-durandin** [comment]
build_jenkins

**Pull request merged**